OMEdit/OMEditGUI/Animation/ViewerWidget.cpp

```diff
@@ -303,8 +363,42 @@ void ViewerWidget::makeShapeInvisible()
       mSelectedShape = "";
     } else {
       shape->setTransparency(1.0);
-      mpAnimationWidget->getVisualizer()->updateVisAttributes(mpAnimationWidget->getVisualizer()->getTimeManager()->getVisTime());
-      mpAnimationWidget->updateScene();
+      mpAnimationWidget->getVisualizer()->modifyShape(mSelectedShape);
+      mSelectedShape = "";
+    }
+  }
+}
+
+/*!
+ * \brief ViewerWidget::changeShapeColor
+ * opens a color dialog and selects a new color for the shape
+ */
+void ViewerWidget::changeShapeColor()
+{
+  ShapeObject* shape = nullptr;
+  if ((shape = mpAnimationWidget->getVisualizer()->getBaseData()->getShapeObjectByID(mSelectedShape))) {
+    if (shape->_type.compare("dxf") == 0)
+    {
+      QString msg = tr("Changing the color is not applicable for DXF-Files.");
+      MessagesWidget::instance()->addGUIMessage(MessageItem(MessageItem::Modelica, "", false, 0, 0, 0, 0, msg, Helper::scriptingKind,
+                                                            Helper::notificationLevel));
+      mSelectedShape = "";
+    }
+    else
+    {
+      QColor currentColor = shape->getColor();
+      QColor color = QColorDialog::getColor(currentColor, this,"Shape Color",  QColorDialog::DontUseNativeDialog);
```

Comment:
Is there a special reason for setting the `QColorDialog::DontUseNativeDialog` flag?